src/overlay/Peer.cpp

```diff
@@ -2009,6 +2007,12 @@ Peer::recvSurveyResponseMessage(StellarMessage const& msg)
 void
 Peer::recvFloodAdvert(StellarMessage const& msg)
 {
+    auto self = shared_from_this();
+    for (auto const& hash : msg.floodAdvert().txHashes)
+    {
+        mApp.getOverlayManager().recvFloodedMsg(self, hash);
```

Comment:
This creates quite of an amplification potential with the current implementation as we bypass flow control here by a factor of 1000. We probably need to either not use the flood map to track adverts (we kinda already do that), or make the floodmap more resilient to DoS (note that `OverlayManager` already has access to both flood map and pull mode related data)